_ml-commons-plugin/algorithms.md

```diff
@@ -0,0 +1,315 @@
+---
+layout: default
+title: Supported Algorithms 
+has_children: false
+nav_order: 100
+---
+
+# Supported Algorithms
+
+ML Commons supports various algorithms to help train and predict machine learning (ML) models or test data-driven predictions without a model. This page outlines the algorithms supported by the ML Commons plugin and the API operations they support.
+
+## Common limitation
+
+Except for the Localization algorithm, all of the following algorithms can only support retrieving 10,000 documents from an index as an input.
+
+## K-Means
+
+K-Means is a simple and popular unsupervised clustering ML algorithm built on top of [Tribuo](https://tribuo.org/) library. K-Means will randomly choose centroids, then calculate iteratively to optimize the position of the centroids until each observation belongs to the cluster with the nearest mean.
+
+### Parameters
+
+Parameter | Type   | Description | Default Value
+:--- |:--- | :--- | :---
+centroids | integer | The number of clusters in which to group the generated data | `2` 
+iterations | integer | The number of iterations to perform against the data until a mean generates | `10`
+distance_type | enum, such as `EUCLIDEAN`, `COSINE`, or `L1` | The type of measurement from which to measure the distance between centroids | `EUCLIDEAN`
+
+### APIs
+
+* [Train]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/#train-model)
+* [Predict]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/#predict)
+* [Train and predict]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/#train-and-predict)
+
+### Example
+
+The following example uses the Iris Data index to train K-Means synchronously. 
+
+```json
+POST /_plugins/_ml/_train/kmeans
+{
+    "parameters": {
+        "centroids": 3,
+        "iterations": 10,
+        "distance_type": "COSINE"
+    },
+    "input_query": {
+        "_source": ["petal_length_in_cm", "petal_width_in_cm"],
+        "size": 10000
+    },
+    "input_index": [
+        "iris_data"
+    ]
+}
```

Comment:
This is an unfair nitpick, but can we indent using spaces instead of tabs? (Most of, if not all) the other code blocks on the docs site are formatted that way, so I want to make sure we're consistent. For reference, lines 91 to 121 go at two spaces with each level of indent.